tools/stonith_admin.c

```diff
@@ -575,25 +575,42 @@ main(int argc, char **argv)
             rc = st->cmds->register_device(st, st_opts, device, NULL, options.agent,
                                            options.params);
             rc = pcmk_legacy2rc(rc);
+            if (rc != pcmk_rc_ok) {
+                out->err(out, "Can't add device: %s",
+                         pcmk_rc_str(rc));
```

Comment:
Especially when this happens in a script we might never see the cmdline so giving the device, topology-level, agent together with the error might be handy.

Reply:
Good point, added a commit on top